presto-spi/src/main/java/com/facebook/presto/spi/type/SqlTimeWithTimeZone.java

```diff
@@ -16,30 +16,64 @@
 import com.fasterxml.jackson.annotation.JsonValue;
 
 import java.time.Instant;
+import java.time.LocalTime;
 import java.time.ZoneId;
+import java.time.ZonedDateTime;
 import java.time.format.DateTimeFormatter;
 import java.util.Objects;
 import java.util.TimeZone;
 
 import static com.facebook.presto.spi.type.DateTimeEncoding.unpackMillisUtc;
 import static com.facebook.presto.spi.type.DateTimeEncoding.unpackZoneKey;
 
+/**
+ * This type represent instant of time in some day and timezone, where timezone
+ * is known and day is not.
+ */
 public final class SqlTimeWithTimeZone
 {
     private static final DateTimeFormatter formatter = DateTimeFormatter.ofPattern("HH:mm:ss.SSS VV");
 
+    /**
+     * As we dont have negative timestamps and max value is midnight in timezone that is ~(UTC -12:00),
```

Comment:
With UTC +12, 23:00:00 equals 1000*60*60*(23-12) = x
With UTC -12, 23:00:00 equals 1000*60*60*(23+12) = y
y > x, or am I wrong with that?

Also, the part about negative millisUtc was wrong. I removed that.